odl/space/npy_tensors.py

```diff
@@ -0,0 +1,2336 @@
+        >>> x = space.element([[1, 2, 3],
+        ...                    [4, 5, 6]])
+        >>> x[0, 1] = -1
+        >>> x
+        rn((2, 3)).element(
+            [[1.0, -1.0, 3.0],
+             [4.0, 5.0, 6.0]]
+        )
+
+        Assignment from array-like structures or another tensor
+        (allowed to be in a different space):
+
+        >>> space = odl.tensor_space((2, 2), dtype='short')
+        >>> y = space.element([[-1, 2],
+        ...                    [0, 0]])
+        >>> x[:, :2] = y
+        >>> x
+        rn((2, 3)).element(
+            [[-1.0, 2.0, 3.0],
+             [0.0, 0.0, 6.0]]
+        )
+        >>> x[0, 1:] = [7, 8]
+        >>> x
+        rn((2, 3)).element(
+            [[-1.0, 7.0, 8.0],
+             [0.0, 0.0, 6.0]]
+        )
+        >>> import numpy as np
+        >>> x[:] = np.array([[0, 0, 0],
+        ...                  [1, 1, 1]])
+        >>> x
+        rn((2, 3)).element(
+            [[0.0, 0.0, 0.0],
+             [1.0, 1.0, 1.0]]
+        )
+
+        Broadcasting is also supported:
+
+        >>> x[:, ::2] = -2.
+        >>> x
+        rn((2, 3)).element(
+            [[-2.0, 0.0, -2.0],
+             [-2.0, 1.0, -2.0]]
+        )
+
+        Array views are preserved:
+
+        >>> y = x[:, ::2]  # view into x
+        >>> y[:] = -9
+        >>> y
+        rn((2, 2)).element(
+            [[-9.0, -9.0],
+             [-9.0, -9.0]]
+        )
+        >>> x
+        rn((2, 3)).element(
+            [[-9.0, 0.0, -9.0],
+             [-9.0, 1.0, -9.0]]
+        )
+
+        Be aware of unsafe casts and over-/underflows, there
+        will be warnings at maximum.
+
+        >>> space = odl.tensor_space((2, 3), 'uint8')
+        >>> x = space.element([[0, 0, 0],
+        ...                    [1, 1, 1]])
+        >>> maxval = 255  # maximum signed 8-bit unsigned int
+        >>> x[0, :] = maxval + 1
+        >>> x
+        tensor_space((2, 3), 'uint8').element(
+            [[0, 0, 0],
+             [1, 1, 1]]
+        )
+        """
+        if isinstance(values, NumpyTensor):
+            self.data[indices] = values.data
+        else:
+            self.data[indices] = values
+
+    @property
+    def real(self):
+        """Real part of ``self``.
+
+        Returns
+        -------
+        real : `NumpyTensor`
+            Real part of this element as a member of a
+            `NumpyTensorSpace` with corresponding real data type.
+
+        Examples
+        --------
+
+        >>> space = odl.cn((2, 3))
+        >>> x = space.element([[1 + 1j, 2, 3 - 3j],
+        ...                    [4, 5 - 5j, 6]])
+        >>> x.real
+        rn((2, 3)).element(
+            [[1.0, 2.0, 3.0],
+             [4.0, 5.0, 6.0]]
+        )
+        """
+        if self.space.is_real_space:
+            return self
+        elif self.space.is_complex_space:
+            # Definitely non-contiguous
+            real_space = self.space.astype(self.space.real_dtype, order='A')
```

Comment:
The point here is that we force `order='A'` instead of inheriting that property from the space. As the comment says (hopefully it's true) we know the result won't be contiguous.